Feature/authentication

Bhdrpkcn opened the pull request:
**### Add Signin and Signup Features, Organize Redux Store, and Improve Login Page Styling**

This pull request adds new features for signin and signup functionality, enhances the organization of Redux store management, and improves the structure of the login page styling. Here are the key highlights:

1. Signin and Signup Features: Implemented signin and signup functionality using Redux Thunk for asynchronous operations. Both features are now managed through the Redux store, providing better organization and avoiding code repetition.

2. Redux Store Management: The signin and signup features are integrated into the Redux store to centralize state management and ensure consistency across the application. This approach improves code maintainability and facilitates future updates.

3. Login Page Styling: The login page styling has been structured to enhance readability and maintainability. While the styling is currently simple, it provides a clear understanding of the components and their functionalities. A simplified version of the mascott animation also added and will be further developed in future updates to create a more engaging user experience. Future iterations will focus on refining the design and adding more advanced styling features.

4. Token Storage: Generated tokens are now stored locally, enhancing security and efficiency by reducing the need for frequent server requests during user sessions.

orabazu commented on package.json:
```diff
@@ -3,18 +3,25 @@
+    "react-router-dom": "^6.21.3",
     "react-scripts": "^5.0.1",
     "react-tinder-card": "^1.6.4",
+    "react-tinder-card": "^1.6.4",
```
react-tinder-card has been added two times

orabazu commented on src/components/body/Current.jsx:
```diff
@@ -0,0 +1,84 @@
+          <h4 className='choresText'>My Tasks</h4>
+          <div className='approvedChores'>
+            {approvedChores.map((chore, index) => (
+              <Card
+                className={`approvedChoresCard ${clickedCard === index ? 'selected' : ''}`}
+                key={`${chore.name}-${chore.name}`}
+                onClick={() => handleCardClick(index)}
+              >
+                <div className='approvedChoresCardContents'>
+                  <div className='approvedChoresCardContentsCred'>
+                    <Card />
+                    {chore.name}
+                  </div>
+                  <div>+ {generatePoints()[0]} points</div>
+                </div>
+                {clickedCard === index && (
+                  <Button
+                    type='primary'
+                    className='additionalButton'
+                    onClick={() => handleRemoveSelectedChore(index)}
+                  >
+                    Remove
+                  </Button>
+                )}
+              </Card>
```
This part could be on its own component

orabazu commented on src/components/body/Current.jsx:
```diff
@@ -0,0 +1,84 @@
+            {approvedChores.map((chore, index) => (
+              <Card
+                className={`approvedChoresCard ${clickedCard === index ? 'selected' : ''}`}
+                key={`${chore.name}-${chore.name}`}
```
```suggestion
                key={`${chore.name}-${index}`}
```

orabazu commented on src/components/body/index.jsx:
```diff
@@ -1,11 +1,60 @@
+      <Route
+        path='/deck'
+        element={
+          token ? (
```
Checkout this simple PrivateRoutes implementation here -> https://medium.com/@dennisivy/creating-protected-routes-with-react-router-v6-2c4bbaf7bc1c

orabazu replied:
Otherwise we'll need to check token within evert `Route`

orabazu commented on src/components/body/login.css:
```diff
@@ -0,0 +1,61 @@
+  height: 40vh;
+}
+
+@keyframes ghostAnimation {
```
Love this 👻

orabazu commented on src/components/body/style.css:
```diff
@@ -1,6 +1,132 @@
+  overflow: hidden;
+}
+
+/* position was absolute, changed to fixed due to overflow problems */
```
unnecessary comment

orabazu commented on src/components/body/style.css:
```diff
@@ -1,6 +1,132 @@
+  background-position: center;
+}
+
+.swipe:last-of-type {
```
empty rule 👀

orabazu commented on src/features/auth/authApi.js:
```diff
@@ -0,0 +1,81 @@
+  'auth/signup',
+  async ({ email, password }, { dispatch }) => {
+    try {
+      const response = await fetchInterceptor(API_URL.SIGNUP, {
```
signup can call plain old `fetch` method here, they shouldn't have any token on client side at this stage

orabazu commented on src/features/auth/authApi.js:
```diff
@@ -0,0 +1,81 @@
+      message.success('Sign-up Successful! Welcome to House Show!')
+      return data
+    } catch (error) {
+      message.error(error.message || 'Sign-up failed. Please try again.')
```
can store this in a const SIGNUP_ERROR = "bla bla"

Bhdrpkcn replied:
i added a constants for  managing error/success messages

orabazu commented on src/features/auth/authApi.js:
```diff
@@ -0,0 +1,81 @@
+  'auth/signin',
+  async ({ email, password }, { dispatch }) => {
+    try {
+      const response = await fetchInterceptor(API_URL.SIGNIN, {
```
same as above for interceptor

orabazu commented on src/components/header/index.jsx:
```diff
@@ -1,13 +1,39 @@
+  useEffect(() => {
+    const storedToken = localStorage.getItem('token')
+    if (storedToken) {
+      dispatch(setCredentials({ user: null, accessToken: storedToken }))
```
we also set accessToken on Login.jsx line:35 is it necessary to call `setCredentials` two times?

orabazu commented on src/components/body/index.jsx:
```diff
@@ -1,11 +1,60 @@
+          </div>
+        }
+      />
+      <Route
```
Login page should redirect to profile when user is logged in already

orabazu commented on src/components/body/index.jsx:
```diff
@@ -1,11 +1,60 @@
+          )
+        }
+      />
+      <Route
```
naming of Current could be Chores or Tasks IMO

Bhdrpkcn commented:
hi @orabazu ;

I've addressed the issues and implemented the necessary changes to improve the codebase.

Here's a brief summary of the key updates:

- PR review fixes
- Refreshing page causing to redirect to '/login' no matter if there is a token exist, also in header signIn/signOut button acting unresponsive. By checking local while refreshing page solved it. 

Would you mind giving it another look when you have a moment? Your feedback is super helpful!

Thanks a bunch! 🚀
